pyproject.toml

```diff
@@ -83,20 +83,30 @@ requires = ["poetry-core>=1.2.0", "poetry-dynamic-versioning"]
 
 [tool.mypy]
 exclude = [
-    '^example\.py$',
-    '^import\.py$',
-    '^tests\/',
     '^doc\/',
     '^ixmp4\/db\/migrations\/',
 ]
 disable_error_code = ['override']
 implicit_reexport = true
-plugins = ['sqlalchemy.ext.mypy.plugin']
+plugins = ['numpy.typing.mypy_plugin', 'pandera.mypy', 'pydantic.mypy', 'sqlalchemy.ext.mypy.plugin']
+disallow_untyped_defs = true
+disallow_any_unimported = true
+# disallow_any_generics = true
```

Comment:
Another note: while working on this PR, I stumbled upon the setting `disallow_any_generics = true`. It is not active as of now, as that would introduce a few hundred more errors, but it would be an even stricter rule we could employ (to follow the spirit of using the strictest possible settings for mypy). This rule would prevent us from returning `dict` or `list` somewhere: all such generics would need to be explicitly specified as `list[int]` or some such. Happy to hear your thoughts on whether we should do that.

Reply:
If we are already at it and its a comparatively small amount of work I would tend to go ahead! "wenn schon denn schon"